library/Isochrone_IGN_API.py

```diff
@@ -18,27 +18,30 @@
 import pandas as pd
 import geopandas as gpd
 import requests
-from shapely.geometry import shape, polygon
+from shapely.geometry import shape, polygon, Point
 from shapely.ops import voronoi_diagram
 from shapely.wkt import loads
 import time
 from .utilsLibrary import decorators
 
 class Isochrone_API_IGN:
-    def __init__(self, input_layer:gpd.GeoDataFrame, range_value:list[int], processingMode:int=0, resource:str='bdtopo-valhalla', costType:str="time", profile:str='car', direction:str='arrival', constraints:str=None, geometryFormat:str='geojson', distanceUnit:str='meter', timeUnit:str='minute', crs:str='EPSG:4326', voronoi_extend_layer=None):
+    def __init__(self, input_layer:gpd.GeoDataFrame, range_value:list[int], processingMode:int=0, resource:str='bdtopo-valhalla', costType:str="time", profile:str='car', direction:str='arrival', constraints:str=None, geometryFormat:str='geojson', distanceUnit:str='meter', timeUnit:str='minute', crs:str='EPSG:4326', voronoi_extend_layer=None,key:str=None):
         """Initializes the Isochrone_API_IGN_V2 class the input layer and the range value.
         Every other parameters are set to their default value but can be changed, they are regrouped in a list to be used in 
         the request_IGN_isochrone_api function.
 
         Args:
             * input_layer (gpd.GeoDataFrame): a GeoDataFrame object representing the location of the points from which the isochrones will be calculated.
             * range_value (list[int]): a list of integers representing the time/distance values for which the isochrones will be calculated.
+            * key (str): The name of the key attribute in the output. If None or 'None', the column will not be created.
             * processingMode (int): Processing mode chosen by the user that decided the output type coming from the Isochrone_API_IGN.main().
             * voronoi_extend_layer (str | tuple): Selected input for the clipping of the voronoi's cells if the voronoi processing mode has been selected.
             * for the other parameters in self.params, see the request_IGN_isochrone_api function
         """
+       
         self.input_layer= input_layer
         self.range_value = range_value
+        self.attributKey = key
         self.processingMode = processingMode
         self.params=[resource,costType,profile,direction,constraints,geometryFormat,distanceUnit,timeUnit,crs]
```

Comment:
```suggestion
        self.params={
            'resource':resource,
            'costType':costType,
            'profile':profile,
            'direction':direction,
            'constraints':constraints,
            'geometryFormat':geometryFormat,
            'distanceUnit':distanceUnit,
            'timeUnit':timeUnit,
            'crs':crs}
```

Not related to your PR but I'm changing it into a dict while I think about it!